dev/nix/shell.nix

```diff
@@ -1,86 +1,87 @@
-{
-  self,
-  inputs,
-  ...
+{ self
+, inputs
+, ...
 }: {
-  perSystem = {
-    inputs',
-    self',
-    pkgs,
-    system,
-    ...
-  }: let
-    isDarwin = pkgs.lib.hasSuffix "darwin" system;
-    fenixPkgs = inputs'.fenix.packages;
-    rustToolchain = with fenixPkgs;
-      fromToolchainFile {
-        file = ../../rust-toolchain.toml;
-        sha256 = "VZZnlyP69+Y3crrLHQyJirqlHrTtGTsyiSnZB8jEvVo=";
-      };
-  in {
-    devShells = {
-      default = pkgs.mkShell {
-        # envs needed for rust toochain
-        RUST_SRC_PATH = "${rustToolchain}/lib/rustlib/src/rust/library";
-        LIBCLANG_PATH = "${pkgs.libclang.lib}/lib";
-        LD_LIBRARY_PATH = "${rustToolchain}/lib";
+  perSystem =
+    { inputs'
+    , self'
+    , pkgs
+    , system
+    , ...
+    }:
+    let
+      isDarwin = pkgs.lib.hasSuffix "darwin" system;
+      fenixPkgs = inputs'.fenix.packages;
+      rustToolchain = with fenixPkgs;
+        fromToolchainFile {
+          file = ../../rust-toolchain.toml;
+          sha256 = "VZZnlyP69+Y3crrLHQyJirqlHrTtGTsyiSnZB8jEvVo=";
+        };
+    in
+    {
+      devShells = {
+        default = pkgs.mkShell {
+          # envs needed for rust toochain
+          RUST_SRC_PATH = "${rustToolchain}/lib/rustlib/src/rust/library";
+          LIBCLANG_PATH = "${pkgs.libclang.lib}/lib";
+          LD_LIBRARY_PATH = "${rustToolchain}/lib";
 
-        # envs needed in order to construct some of the rust crates
-        ROCKSDB_LIB_DIR = "${pkgs.rocksdb}/lib/";
-        OPENSSL_NO_VENDOR = 1;
-        OPENSSL_DIR = "${pkgs.openssl.dev}";
-        OPENSSL_INCLUDE_DIR = "${pkgs.openssl.dev}/include";
-        OPENSSL_LIB_DIR = "${pkgs.openssl.out}/lib";
-        packages = with pkgs; [
-          # core tooling to share across linux/macos
-          coreutils
-          pkg-config
-          protobuf
-          libiconv
-          openssl
-          gnumake
+          # envs needed in order to construct some of the rust crates
+          ROCKSDB_LIB_DIR = "${pkgs.rocksdb}/lib/";
+          OPENSSL_NO_VENDOR = 1;
+          OPENSSL_DIR = "${pkgs.openssl.dev}";
+          OPENSSL_INCLUDE_DIR = "${pkgs.openssl.dev}/include";
+          OPENSSL_LIB_DIR = "${pkgs.openssl.out}/lib";
+          packages = with pkgs; [
+            # core tooling to share across linux/macos
+            coreutils
+            pkg-config
+            protobuf
+            libiconv
+            openssl
+            gnumake
 
-          # tools for e2e testing
-          docker-compose
-          python311
-          python311Packages.pip
-          python311Packages.virtualenv
-          sops
+            # tools for e2e testing
```

Comment:
This comment has confusion factor 100.

`docker-compose` is not needed by `e2e-tests` per se. It is at most needed for running local environment.

`python312` from devshell is not suitable for running `e2e-tests`, because numpy won't work in it.

Reply:
I agree. The diff makes it look like I added the comment - i didn't

Reply:
`numpy` issue is not a global issue, it's caused by file system is missing something. I can try to replicate that @LGLO if you give me your specs. AFAIK 3.12 works fine, both in devshell and native installations, both on mac and linux.